extension/intents/slideshow/contentScript.js

```diff
@@ -0,0 +1,347 @@
+      tagViewSlide.textContent = String.fromCharCode(10696);
+      tagViewSlide.onclick = toggleGallery;
+
+      const tagViewGallery = document.createElement("a");
+      tagViewGallery.className = "fv-view-gallery";
+      tagViewGallery.textContent = String.fromCharCode(9638);
+      tagViewGallery.onclick = toggleGallery;
+
+      // images and video container
+      slideContainer = iframeDoc.createElement("div");
+      slideContainer.className = "fv-slide-image";
+
+      // putting it all together
+      slideshowContainer.append(slideContainer);
+
+      navbar.appendChild(tagViewGallery);
+      navbar.appendChild(tagViewSlide);
+      navbar.append(tagClose);
+      navbar.append(tagNext);
+      navbar.append(tagPrev);
+
+      galleryContainer.appendChild(thumbnailGallery);
+
+      lightboxElement.append(navbar);
+      lightboxElement.append(slideshowContainer);
+      lightboxElement.appendChild(galleryContainer);
+
+      iframeDoc.body.append(lightboxElement);
+
+      // find images and videos on current tab
+      slideElements = detectAllMedia();
+
+      // generate thumbnail gallery elements
+      buildThumbnailGallery();
+
+      // join gallery elements to build gallery
+      galleryElements.forEach(function(element) {
+        thumbnailGallery.append(element);
+      });
+
+      // make iframes responsive
+      window.addEventListener("resize", function() {
+        iframeContainer.height = document.documentElement.clientHeight;
+        iframeContainer.width = document.documentElement.clientWidth;
+        if (slideContainer.hasChildNodes()) {
+          if (slideContainer.firstChild.tagName === "IFRAME") {
+            slideContainer.firstChild.setAttribute(
+              "width",
+              document.documentElement.clientWidth
+            );
+            slideContainer.firstChild.setAttribute(
+              "height",
+              document.documentElement.clientHeight
+            );
+          }
+        }
+      });
+    });
+  }
+
+  function hideSlideShow(event) {
+    if (event.target.className === "fv-close") {
+      iframeContainer.style.display = "none";
+      swapSlide();
+    }
+  }
+
+  function revealSlideshow() {
+    swapSlide(slideElements[0]);
+    iframeContainer.style.display = "block";
+  }
+
+  function isVideoSourceUrl(url) {
+    let testExp = "";
+    const result = videoSources.some(function(videoSource) {
+      testExp = new RegExp(`^(https?:\/\/)?(www\.)?${videoSource}`);
+      return testExp.test(url);
+    });
+
+    return result;
+  }
+
+  function detectAllMedia() {
+    const selector = "img, video, iframe";
+    // detect all supported elements
+    const mediaElements = document.body.querySelectorAll(selector);
+
+    // iterate thru list and add to array
+
+    mediaElements.forEach(element => {
+      // strip their current classes
+
+      element.className = "";
+      switch (element.tagName) {
+        case "IMG": {
```

Comment:
I think there should be a minimum size for images (probably looking at clientHeight/Width). E.g., at least 200px in both directions. As it is now it will pick up some controls and icons.

Reply:
Set a minimum size for images of 200x200px